### PR TITLE
docs: moved the policy regarding commits from when it was running locally to the contributor page

### LINKED ISSUE #11700: Relocation of Commit Guidelines Information in the Documentation
# Summary

What change needs making?

I believe that information about commits is essential for contributors when it comes to contributing. However, details about commits were not directly accessible in the contribution README file; they could only be viewed by accessing the "running locally" file. To make contributing to the Argo Workflows project more accessible, I've moved the commit rules information to the contribution page. Additionally, I've added extra links related to commit rules.

image

image

# Use Cases

When would you use this?

Someone attempting to contribute to the Argo Workflows project for the first time can easily find the project's commit rules, facilitating their contribution process.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by sunyeongchoi**
Fixes #11700

### Modifications

moved the policy regarding commits from when it was running locally to the contributor page and I have also improved some links related to commits

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +35,36 @@ We're always looking for contributors.
...
+
+```bash
+git commit --signoff -m 'feat: Added a new feature. Fixes #1234'
+```
+
+TroubleShooting:
+
+* When running `make pre-commit -B`, if you encounter errors like
+  `make: *** [pkg/apiclient/clusterworkflowtemplate/cluster-workflow-template.swagger.json] Error 1`, ensure that you
+  have checked out your code into `$GOPATH/src/github.com/argoproj/argo-workflows`.
```
Comment (terrytangyuan) [inline docs/CONTRIBUTING.md:66]: But this pre-commit requires installing the toolchain which are listed in running-locally document.

Comment (agilgur5) [reply docs/CONTRIBUTING.md:66]: Yea this may make more sense to just link out to `running-locally.md#committing`

Comment (sunyeongchoi) [reply docs/CONTRIBUTING.md]: @terrytangyuan @agilgur5 

Thank you for your review! :)

Based on your feedback, I have kept the Committing-related information in the "running locally" file as is, and added a link to the Committing section within the "running locally" file to the "contributing" file.

This is expected to make it easier for developers to find the commit guidelines for PRs.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ We're always looking for contributors.
 
 To run Argo Workflows locally for development: [running locally](running-locally.md).
 
+### Committing
+
+To commit Argo Workflows for PR: [Committing](running-locally.md#committing).
```
Comment (agilgur5) [inline docs/CONTRIBUTING.md:40]:
```suggestion
See the [Committing Guidelines](running-locally.md#committing).
```

**Review by agilgur5 — COMMENTED** [review]
Works for me.
An alternative may be to add the guidelines in `CONTRIBUTING.md`, then link to Running Locally for running pre-commit.

Do you want to add the links you added before into the Running Locally page?

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (sunyeongchoi) [reply docs/CONTRIBUTING.md:40]: @agilgur5 

I added a commit adding a link to signoff.
And your suggestion is good :)
Thank you for your sincere review!

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**